File: 442. Find All Duplicates in an Array/Solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        nums.sort()
        r=[]
        for i in range(1,len(nums)):
            if nums[i]==nums[i-1]:
                if not r or r[-1] != nums[i]:
                    r.append(nums[i])
        return r

File: 442. Find All Duplicates in an Array/Solution.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> r;
        sort(nums.begin(),nums.end());
        for(int i=1; i<nums.size(); i++){
            if(nums[i] == nums[i-1]){
                if(r.empty() || r.back() != nums[i]){
                    r.push_back(nums[i]);
                }
            }
        }
        return r;
    }
};
